### ISSUE TITLE
Schema

### ISSUE THREAD

**Issue by dazzaji** [mitmedialab/open-us-legal-corpus-creator#2]
Consider adding "date" to schema to capture, as closely as possible, the date (or at least the year) a given statute became legally binding (aka the "sunrise" date).  

Here is the legacy schema from the Australian version, which we should use as a starting point:

version_id
type
jurisdiction
source
citation
url
text

I'm well aware this is among the deepest rabbit holes in the entire field of law, with confounding factors like how to deal with amendment dates and so many other wrinkles.  But law.MIT.edu does eventually want to evolve this into a more dynamic temporal expression of the law, and perhaps starting with one date statement will help push the ETL toward doing more complex capturing of date and time and duration and related information in future versions.

**Comment by dazzaji** [comment]
I grabbed the schema from here, but the way: https://huggingface.co/datasets/umarbutler/open-australian-legal-corpus

**Comment by umarbutler** [comment]
Here's my proposed schema, noting that the only real differences between it and the schema employed by the Open Australian Legal Corpus ('OALC') are the inclusion of `time` and `downloaded` fields:
| Field | Type | Description |
| - | - | - |
| version_id | string | A unique, persistent and ideally semantic identifier for the latest known version of the document, formatted as `'{source}:{id}'`, preferably one employed by the source itself. |
| type | string | The type of the document. Possible values are `primary_legislation`, `secondary_legislation`, `bill` and `decision`. |
| jurisdiction | string | The jurisdiction of the document, in lowercase, hyphen-delimited form, with `federal` used for the federal jurisdiction. |
| source | string | The source of the document. |
| time | string | The RFC 3339 formatted UTC time at which the latest known version of the document was published, approved or entered into force, in descending order of priority. |
| citation | string | The title of the document. |
| url | string | A hyperlink to the latest known version of the document, ideally as persistent and authoritative as possible, with persistency prioritised above authoritativeness. |
| downloaded | string | The RFC 3339 formatted UTC time at which the latest known version of the document was downloaded. |
| text | string | The text of the latest known version of the document. |

**Comment by kazimuth** [comment]
I think that makes sense. 

It might be worth having separate time fields for published / approved / entered into force, if only because while we're writing scrapers that information may be obvious, and it seems a shame to throw it out if we do have it in some situations.

We can always leave the fields blank, and unify them into a single time field like above when doing analysis.

**Comment by umarbutler** [comment]
> It might be worth having separate time fields for published / approved / entered into force, if only because while we're writing scrapers that information may be obvious, and it seems a shame to throw it out if we do have it in some situations.

While I do see the value in this approach given  the difficulties we might encounter if we tried to add more fields at a later date, I think we should also try to keep things as simple as possible in the first instance and avoid compounding our work considering our limited capacity.

Have a look at [this](https://supreme.justia.com/cases/federal/us/60/393/) random US Supreme Court case I found, there are multiple dates of arguments and a different date of the decision which has a different year to what appears in the case’s citation. I also know that Australian court cases can have a hearing date, judgement date, publication date and amendement date. Statutes can have the date the were passed, received royal or presidential assent, were published on the Federal Register and came into force.

One thing we could do which you’ve given me an idea for is to have a specifying what type of time we’ve provided. So if we use the publication date for a particular document as its time, we can put ‘publication’ in that field. We could call it `time_type`.

Also, as we go along, if we do find that it would be easy for us to add more specific temporal fields, than we could do that, but perhaps for now it may be preferable to keep things simple.

Anyways happy to get your further thoughts on this  @kazimuth @dazzaji
